packages/@azure/servicebus/data-plane/lib/core/messageReceiver.ts

```diff
@@ -946,8 +941,12 @@ export class MessageReceiver extends LinkEntity {
         );
         // It is possible for someone to close the receiver and then start it again.
         // Thus make sure that the receiver is present in the client cache.
-        if (this.receiverType === ReceiverType.streaming && !this._context.streamingReceiver) {
-          this._context.streamingReceiver = this as any;
+        if (this.receiverType === ReceiverType.streaming) {
+          if (!this._context.streamingReceiver) {
+            this._context.streamingReceiver = this as any;
+          } else {
+            this._context.streamingReceiver.name = this._receiver.name;
```

Comment:
Why do we need this change?

Reply:
When recovering from a network connection loss, we create a new receiver but we are not updating new receiver name anywhere in `this.context`.
When we call `await brokeredMessage.complete();` It calls `this._context.getReceiver()` method where we check context receiver name is equals to the new receiver name. If not then we throw `Cannot find the receiver with name '${name}'` error.
I was thinking to update the whole receiver into the context but If we do that it will not update the old receivers properties(like receive settle mode etc) in new receiver.

So I just updated the new receiver name in the context receiver. It works fine now without any error.

Reply:
As discussed offline, lets update the name right after we decide what the name for the new receiver should be